src/move/econia/sources/market.move

```diff
@@ -3044,7 +3241,9 @@ module econia::market {
             base_available, base_ceiling, quote_available, quote_ceiling);
         // Assume no assets traded as a taker.
         let (base_traded, quote_traded, fees) = (0, 0, 0);
-        if (crosses_spread) { // If order price crosses spread:
+        // Evaluate any potential fills across the spread, which may
+        // result in deferred fill events:
```

Comment:
I'm confused why we have this queue instead of always emitting the event directly to the handle?

Reply:
When a limit order takes and then makes, the assigned order ID needs to encode a pointer into AVL queue memory that is only assigned upon insertion to the order book (the pointer, or "AVL queue access key" is used to lookup on order size change or order cancels).

But, when a limit order takes then makes, the take portion (in matching engine) mutates the AVL queue and so the AVL queue access key is not known until the make portion that posts. Hence, to enforce a common order ID, the events are deferred until after the match operation. See the changes to `place_limit_order()`